### PR TITLE
amrfinder: new package

### PR DESCRIPTION
**Pull request by luke-dt**

### CODE REVIEW

File: var/spack/repos/builtin/packages/amrfinder/package.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+class Amrfinder(Package):
+    """NCBI AMRFinderPlus: This software and the accompanying database identify
+       acquired antimicrobial resistance genes in bacterial protein and/or
+       assembled nucleotide sequences as well as known resistance-associated
+       point mutations for several taxa. With AMRFinderPlus we added select
+       members of additional classes of genes such as virulence factors,
+       biocide, heat, acid, and metal resistance genes."""
+
+    homepage = "https://github.com/ncbi/amr/wiki"
+    url      = "https://github.com/ncbi/amr/releases/download/amrfinder_v3.10.24/amrfinder_binaries_v3.10.24.tar.gz"
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/amrfinder/package.py]: Why installing binaries, when sources seem to be available?

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/amrfinder/package.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    def install(self, spec, prefix):
+        mkdir(prefix.bin)
+        install('amrfinder', prefix.bin)
+        install('amrfinder_update', prefix.bin)
+        install('amr_report', prefix.bin)
+        install('dna_mutation', prefix.bin)
+        install('fasta2parts', prefix.bin)
+        install('fasta_check', prefix.bin)
+        install('fasta_check', prefix.bin)
+        install('gff_check', prefix.bin)
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/amrfinder/package.py]: This method needs to be updated too.

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/amrfinder/package.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+    depends_on('blast-plus')
+    depends_on('hmmer')
+    depends_on('curl')
+
+    def edit(self, spec, prefix):
+        mkdirp(prefix.bin)
+        mkdirp(prefix.share)
+        env['INSTALL_DIR'] = prefix.bin
+        env['DEFAULT_DB_DIR'] = prefix.share
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/amrfinder/package.py]: Last minor request.We are overriding the `edit` function, but we are effectively not doing any editing. I think:
```suggestion
    def setup_build_environment(self, env):
        env.set('INSTALL_DIR', prefix.bin)
        env.set('DEFAULT_DB_DIR', prefix.share)

    @run_before('build')
    def create_bin_and_share(self):
        mkdirp(self.spec.prefix.bin)
        mkdirp(self.spec.prefix.share)
```
Does this work too?

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/amrfinder/package.py (hunk 3 above)
Comment (luke-dt) [reply var/spack/repos/builtin/packages/amrfinder/package.py]: Confirmed that this works and the database downloads to the correct location

**Review by alalazo — APPROVED** [review]

**Pull request merged**